BootloaderCommonPkg/Include/Library/FirmwareUpdateLib.h

```diff
@@ -336,31 +427,35 @@ FirmwareUpdateGetComponentInfo (
   This function will get state machine flag from the bootloader reserved region
   First byte in the booloader reserved region is state machine flag
 
-  @param[in, out] StateMachine  Pointer to state machine flag byte.
+  @param[in]      FwUpdStatusOffset Offset to the fw update status structure in reserved region.
+  @param[in, out] StateMachine      Pointer to state machine flag byte.
 
   @retval  EFI_SUCCESS        State machine flag found.
   @retval  others             Error while getting state machine flag.
 **/
 EFI_STATUS
 GetStateMachineFlag (
+  IN     UINT32   FwUpdStatusOffset,
```

Comment:
I saw PwUpdStatusOffset get passed in everywhere.  In this case, I would recommend to use a module patchable PCD to do it so that you don't have to add new argument in the interfaces.

Reply:
Added new PCD PcdFwUpdStatusBase